tuist/tuist
# Prevent embedding static frameworks

Resolves https://github.com/tuist/tuist/issues/454

### Short description 📝

Precompiled static frameworks were being embedded when generating projects.

### Solution 📦

- Ensure the linkage of precompiled frameworks is checked to determine if they need to be embedded

### Notes 📝:

- All `FrameworkNodes` (i.e. prebuilt frameworks) were being embedded regardless of their linkage
- This check was redundant as there was a more general check for `PrecompiledNode` which happens to be the super class of `FrameworkNode` which took into account the framework linkage
- To resolve this issue it was suffice to remove the redundant check

### Implementation 👩‍💻👨‍💻

- [x] Update Graph embeddable references logic


### Test Plan 🛠:

- Generate `fixtures/ios_app_with_static_frameworks` via running `tuist generate`
- Verify the generated project doesn't embed `PrebuiltStaticFramework`
- Verify the generated project builds successfully
